File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <stack>
using namespace std;

string lines;
stack<char> s;

int main() {
    cin >> lines;

    int ans = 0;

    for (int i = 0; i < lines.size(); i++) {
        if (lines[i] == '(') {
            s.push('(');
        } else {
            if (lines[i - 1] == '(') {
                s.pop();
                ans += s.size();
            } else {
                s.pop();
                ans += 1;
            }
        }
    }

    cout << ans;

    return 0;
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
import sys
input = sys.stdin.readline

str = input().rstrip()

stack = []
tmp = 0
answer = 0
for i in range(len(str)):
    if str[i] == "(":
        stack.append(str[i])
    else:
        if str[i - 1] == "(": # 레이저
            stack.pop()
            answer += len(stack)
        else:
            stack.pop()
            answer += 1
     
print(answer)